browser/src/control/Control.UIManager.js

```diff
@@ -917,6 +917,55 @@ L.Control.UIManager = L.Control.extend({
 		], cancelButtonId);
 	},
 
+	/// buttonObjectList: [{id: button's id, text: button's text, ..other properties if needed}, ...]
+	/// callbackList: [{id: button's id, func_: function}, ...]
+	showModalWithCustomButtons: function(id, title, message, cancellable, buttonObjectList, callbackList) {
```

Comment:
maybe "yes/no" modal can now use this function internally

Reply:
I guess no because yesno modal only has 2 buttons. This one can have more than 2 buttons.

Reply:
hmm I don't see any limitation here, it just uses array of button objects

Reply:
Right, i mis-read your comment at first :) we can shorten that part indeed.